BindingContext/BPoP.md

```diff
@@ -0,0 +1,334 @@
+
+```
+BPoP: enabled, SameSite=None
+```
+
+The latest configuration replaces the previous configuration. Replacement occurs for the entire config, not just for configuration elements who appear in the `BPoP` header.
+
+Please note that this behavior applies only when there is no enterprise policy override. In case if enterprise policy enabled by the browser config, the BPoP behaves as configured by the policy. More details on key management and storage for enterprise use cases are covered later in this document.
+
+### Retrieval model
+
+The browser only attaches BPoP proofs to "secure" protocols (as defined by the user agent).
+
+The browser maintains one public private key pair per sub entity to the effective top-level domain (eTLD+1). That is, if `a.example.com`, `b.example.com`, and `c.example.com` each activate BPoP, they share a single public private key pair.
+
+The browser attaches BPoP proofs to a request if there exists a config that either:
+
+- Has an exact match between the BPoP origin and the canonicalized host of the retrieval's URI and the `subdomains` flag is false
+- Has a domain match between the BPoP origin and the canonicalized host of the retrieval's URI and the `subdomains` flag is true
+
+The semantics of `SameSite` match the cookie attributes. That is, if the browser would not attach a cookie with SameSite=Lax to a request, and the server has initialized BPoP with SameSite=Lax, the browser should not attach a BPoP proof to the request.
+
+TBD: Add how `browser-policy` is defined and how it will impact the key management.
+
+### Application model
+
+#### BPoP key verification
+
+The application generated cookie bound by BPoP should not need to verify the asymmetric key everytime. Instead, the application should cache the `key-verification` status and should be able to request the key verification only when the cache is expired or invalidated. Cache validity can be equivalent to `BPoP-Nonce` validity.
+
+The browser initiates BPoP proof of possession only if:
+
+- BPoP nonce is expired or invalidated; set cache expiration time to `BPoP-Nonce` expiration time.
+
+```
+BPoP: enabled, expires-in=300
+```
+
+#### BPoP background refresh
+
+The server response should also include a `refresh-in` parameter, that enables the app to refresh the nonce in a certain interval. This should help the APIs dependent on BPoP to proactively refresh the nonce in the background avoiding the need to refresh the nonce on demand.
+
+```
+BPoP: enabled, refresh-in=60
+```
+
+### Server challenge
+
+A server may reject a BPoP proof because its nonce is missing or out of date:
+
+```
+HTTP/1.1 401 Unauthorized
+WWW-Authenticate: BPoP error="use_bpop_nonce", error_description="Web server requires nonce in BPoP proof"
+BPoP-Nonce: eyJ7S_zG.eyJH0-Z.HX4w-7v
+```
+
+In such a case, the browser should update the BPoP proof and retry the request immediately. Servers SHALL not ask for more than one retry this way.
```

Comment:
If the same client sends multiple concurrent requests in flight with the same BPoP proof before it receives a `401` and `WWW-Authenticate: BPoP...` response from the server, how should the server respond to the other requests that were already sent with the same invalid BPoP proof?

A comparable [DBSC](https://github.com/WICG/dbsc/blob/main/README.md#maintaining-a-session) proposal attempts to solve this by client-side logic in the protocol that holds and queues requests when its proof has expired until it can refresh it (it uses an auxiliary cookie as its proof).